src/regexp.js

```diff
@@ -69,87 +114,443 @@ const fuzzPrintableAscii = f => {
   return String.fromCharCode(32 + f.rng.nextInt(94));
 }
 
+const fuzzHexExcept = (f, except) => oneOf(...'01234567890abcdefABCDEF'.split('').filter(hex => except.indexOf(hex) === -1))(f);
+
 const fuzzHex = oneOf('0','1','2','3','4','5','6','7','8','9','a','b','c','d','e','f','A','B','C','D','E','F');
 
+const decimal = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'];
 
-const fuzzAlternation = f => {
-  if (f.tooDeep()) return '';
-  f = f.goDeeper();
-  return many(choose(fuzzGrouping, fuzzCharacterClass, fuzzRepetition, fuzzSequence))(f).join('|');
+const syntaxCharacters = '^$\\.*+?()[{|'.split('');
+
+const unicodeSyntaxCharacters = '^$\\.*+?()[]{}|';
+
+const controlLetters = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ'.split('');
+
+const getSyntaxCharacters = state => state.unicode ? unicodeSyntaxCharacters : syntaxCharacters;
+
+const utf16GeneralCategoryValues = ['Cased_Letter', 'LC', 'Close_Punctuation', 'Pe', 'Connector_Punctuation', 'Pc', 'Control', 'Cc', 'cntrl', 'Currency_Symbol', 'Sc', 'Dash_Punctuation', 'Pd', 'Decimal_Number', 'Nd', 'digit', 'Enclosing_Mark', 'Me', 'Final_Punctuation', 'Pf', 'Format', 'Cf', 'Initial_Punctuation', 'Pi', 'Letter', 'L', 'Letter_Number', 'Nl', 'Line_Separator', 'Zl', 'Lowercase_Letter', 'Ll', 'Mark', 'M', 'Combining_Mark', 'Math_Symbol', 'Sm', 'Modifier_Letter', 'Lm', 'Modifier_Symbol', 'Sk', 'Nonspacing_Mark', 'Mn', 'Number', 'N', 'Open_Punctuation', 'Ps', 'Other', 'C', 'Other_Letter', 'Lo', 'Other_Number', 'No', 'Other_Punctuation', 'Po', 'Other_Symbol', 'So', 'Paragraph_Separator', 'Zp', 'Private_Use', 'Co', 'Punctuation', 'P', 'punct', 'Separator', 'Z', 'Space_Separator', 'Zs', 'Spacing_Mark', 'Mc', 'Surrogate', 'Cs', 'Symbol', 'S', 'Titlecase_Letter', 'Lt', 'Unassigned', 'Cn', 'Uppercase_Letter', 'Lu']
```

Comment:
Weren't we going to rely on @mathiasbynens's library for these?

Reply:
This feature has been moved into the `regexp-es2018` branch (no open PR yet), but I will use his library in that branch.